### PR TITLE
added request form

### PR DESCRIPTION
**Pull request by Tomer51m**
added request form:
this has no css at the moment. only html skeleton.
added route to request page and all form functionality except shared state between components.
I will add the option to go to previous form and edit state in the next commit.

### CODE REVIEW

File: src/components/ProjectRequestForm/FormConifrmation.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        please confirm form details:
+        {
+            Object.entries(formState).map(([key, value]) => {
+                return <div>{key}: {value}</div>
+            })
+        }
+        <button onClick={loadPreviousForm}>Previous</button>
+        <button type="submit">Submit</button>
+      </form>
+    </div>
```
Comment (Polarts) [inline src/components/ProjectRequestForm/FormConifrmation.jsx:21]: Is there a point to wrap a single form element in a div? You can return a form directly in your statement.

File: src/components/ProjectRequestForm/FormForProfit.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import styles from "./requestForm.module.scss";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
```
Comment (Polarts) [inline src/components/ProjectRequestForm/FormForProfit.jsx:4]: Is Yup a component? If not. it should start with lower case.

File: src/components/ProjectRequestForm/FormForProfit.jsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          />
+
+          <label className={styles.addFileLabel} htmlFor="systemDefinitionFile">
+            Select a file:
+          </label>
+          <input
+            type="file"
+            id="systemDefinitionFile"
+            name="systemDefinitionFile"
+          ></input>
```
Comment (Polarts) [inline src/components/ProjectRequestForm/FormForProfit.jsx:95]: File should be provided via external link.

File: src/components/ProjectRequestForm/FormUser.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          <Field name="fullName" type="text" />
+          <ErrorMessage
+            name="fullName"
+            render={(msg) => <span className={styles.formError}>{msg}</span>}
+          />
+        </div>
+
+        <div className={styles.inputField}>
+          <label htmlFor="email">Email:</label>
+          <Field name="email" type="text" />
```
Comment (Polarts) [inline src/components/ProjectRequestForm/FormUser.jsx:47]: Email inputs should be of type `email`

File: src/components/ProjectRequestForm/FormConifrmation.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+export default function FormConifrmation({ formState, loadPreviousForm }) {
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    alert(JSON.stringify(formState, null, 2));
```
[the remark this answers is not in the document]
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormConifrmation.jsx:6]: This is just to preview the ready object - until sergey sets up the routes in the server

File: src/components/ProjectRequestForm/FormConifrmation.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        please confirm form details:
+        {
+            Object.entries(formState).map(([key, value]) => {
+                return <div>{key}: {value}</div>
+            })
+        }
```
[the remark this answers is not in the document]
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormConifrmation.jsx:17]: First version of the component. the new commit is fixed.

File: src/components/ProjectRequestForm/FormConifrmation.jsx (hunk 1 above)
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormConifrmation.jsx:21]: You're right. Fixed

File: src/components/ProjectRequestForm/FormForProfit.jsx (hunk 2 above)
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormForProfit.jsx:4]: I saw few ways this is being done online - changed to lowercase.

File: src/components/ProjectRequestForm/FormForProfit.jsx (hunk 3 above)
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormForProfit.jsx:95]: ok. changed

File: src/components/ProjectRequestForm/FormUser.jsx (hunk 4 above)
Comment (haifa-dev) [reply src/components/ProjectRequestForm/FormUser.jsx:47]: fixed

**Pull request merged**